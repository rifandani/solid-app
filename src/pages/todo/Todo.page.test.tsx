import { Route } from '@solidjs/router';
import { screen, waitFor } from '@solidjs/testing-library';
import { mockedCreateResource } from '../../mocks/module.mock';
import { renderProviders } from '../../utils/test.util';
import TodoPage from './Todo.page';

describe('TodoPage', () => {
  it('should render correctly', async () => {
    // ARRANGE
    const { container } = renderProviders(() => <Route path="/" component={TodoPage} />);

    // ASSERT
    await waitFor(() => {
      expect(mockedCreateResource).toHaveBeenCalled();
      expect(screen.getByText(/Todo List/)).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
