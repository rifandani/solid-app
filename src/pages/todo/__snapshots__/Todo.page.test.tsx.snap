// Vitest Snapshot v1

exports[`TodoPage > should render correctly 1`] = `
<main
  class="py-20 px-10 md:px-24 lg:px-40 xl:px-52 flex flex-col items-center justify-center duration-300"
>
  <h1
    class="text-2xl mb-10 font-semibold tracking-wider text-violet-500"
  >
    Todo List
  </h1>
  <section
    class="block w-full border rounded p-5 shadow-md border-slate-300"
  >
    <form
      class="flex flex-col w-full mb-5 duration-300 md:flex-row"
    >
      <input
        autofocus=""
        class="p-2 w-10/12 mr-2 duration-300"
        id="todoText"
        name="todoText"
        placeholder="What should you do next..."
        required=""
        type="text"
      />
      <button
        class="font-semibold items-center justify-center inline-flex space-x-2 select-none cursor-pointer duration-200 text-white bg-violet-800 border-0 ring-violet-800 hover:bg-violet-500 active:bg-violet-200 focus:ring-4 disabled:opacity-50 rounded text-md py-4 w-2/12 mt-2 px-0.5 md:mt-0"
        type="submit"
      >
        Add Todo
      </button>
    </form>
    
    <div
      class="flex py-5 justify-center items-center"
    >
      No Data
    </div>
  </section>
</main>
`;

exports[`TodoPage > should render correctly 2`] = `
<main
  class="flex flex-col items-center justify-center py-20 px-10 duration-300 md:px-24 lg:px-40 xl:px-52"
>
  <h1
    class="mb-10 text-2xl font-semibold tracking-wider text-primary-content"
  >
    Todo List
  </h1>
  <section
    class="card w-full rounded-lg border bg-secondary p-5 text-secondary-content shadow-lg"
  >
    <form
      class="form-control  mb-3 w-full duration-300 lg:flex-row"
    >
      <input
        autofocus=""
        class="input-bordered input-accent input w-full text-accent-content lg:w-10/12"
        id="todoText"
        name="todoText"
        placeholder="What should you do next..."
        required=""
        type="text"
      />
      <button
        class="btn-accent btn mt-2 ml-0 w-full normal-case lg:mt-0 lg:ml-2 lg:w-2/12"
        type="submit"
      >
        Add âœ”
      </button>
    </form>
    
    <form
      class="mb-3 flex w-full flex-col duration-300 md:flex-row md:space-x-2"
    >
      <label
        class="label"
      >
        <span
          class="label-text"
        >
          Sort by
        </span>
      </label>
      <select
        class="select-bordered select select-md"
      >
        <option
          value="newest"
        >
          Newest
        </option>
        <option
          value="oldest"
        >
          Oldest
        </option>
      </select>
      <label
        class="label"
      >
        <span
          class="label-text"
        >
          Filter by
        </span>
      </label>
      <select
        class="select-bordered select select-md"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="completed"
        >
          Completed
        </option>
        <option
          value="incomplete"
        >
          Incomplete
        </option>
      </select>
    </form>
    <div
      class="flex items-center justify-center py-5"
    >
      No Data
    </div>
  </section>
</main>
`;
